src/Microsoft.CodeAnalysis.Analyzers/CSharp/CSharpUpgradeMSBuildWorkspaceAnalyzer.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Analyzers;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class CSharpUpgradeMSBuildWorkspaceAnalyzer : UpgradeMSBuildWorkspaceAnalyzer
+    {
+        private CSharpUpgradeMSBuildWorkspaceAnalyzer(bool performAssemblyChecks)
+            : base(performAssemblyChecks)
+        {
+        }
+
+        public CSharpUpgradeMSBuildWorkspaceAnalyzer()
+            : this(performAssemblyChecks: true)
+        {
+        }
+
+        internal static CSharpUpgradeMSBuildWorkspaceAnalyzer CreateForTests()
+            => new CSharpUpgradeMSBuildWorkspaceAnalyzer(performAssemblyChecks: false);
+
+        protected override void RegisterIdentifierAnalysis(CompilationStartAnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(AnalyzerIdentifier, SyntaxKind.IdentifierName);
+        }
+
+        protected override void RegisterIdentifierAnalysis(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(AnalyzerIdentifier, SyntaxKind.IdentifierName);
+        }
+
+        private void AnalyzerIdentifier(SyntaxNodeAnalysisContext context)
```

Comment:
Did you mean to call it AnalyzeIdentifier?

Reply:
oops :smile: